cql3/expr/prepare_expr.cc

```diff
@@ -933,13 +916,7 @@ prepare_expression(const expression& expr, data_dictionary::database db, const s
             return null_prepare_expression(db, keyspace, receiver);
         },
         [&] (const bind_variable& bv) -> expression {
-            switch (bv.shape) {
-            case expr::bind_variable::shape_type::scalar:  return bind_variable_scalar_prepare_expression(bv, db, keyspace, receiver);
-            case expr::bind_variable::shape_type::scalar_in: return bind_variable_scalar_in_prepare_expression(bv, db, keyspace, receiver);
-            case expr::bind_variable::shape_type::tuple: on_internal_error(expr_logger, "prepare_expression(bind_variable(tuple))");
-            case expr::bind_variable::shape_type::tuple_in: on_internal_error(expr_logger, "prepare_expression(bind_variable(tuple_in))");
-            }
-            on_internal_error(expr_logger, "unexpected shape in bind_variable");
+            return bind_variable_scalar_prepare_expression(bv, db, keyspace, receiver);
```

Comment:
Very good change. It's like my previous comment regarding caller context in the callee: previously, bind_valiable_*_prepare_expression had to know about the caller context (communicated by calling different functions), and now the callee doesn't care and the caller configures it by setting the correct receiver type.